internal/provider/providers/nowdns/provider.go

```diff
@@ -0,0 +1,171 @@
+}
+
+func (p *Provider) Domain() string {
+	return p.domain
+}
+
+func (p *Provider) Host() string {
+	return p.host
+}
+
+func (p *Provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *Provider) Proxied() bool {
+	return false
+}
+
+func (p *Provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *Provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName()),
+		Host:      p.Host(),
+		Provider:  "<a href=\"https://www.now-dns.com/\">Now-DNS</a>",
+		IPVersion: p.ipVersion.String(),
+	}
+}
+
+func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Addr) (newIP netip.Addr, err error) {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "now-dns.com",
+		Path:   "/update",
+		User:   url.UserPassword(p.username, p.password),
+	}
+
+	values := url.Values{}
+	values.Set("hostname", utils.BuildURLQueryHostname(p.host, p.domain))
+	if !p.useProviderIP {
+		values.Set("myip", ip.String())
+	}
+	u.RawQuery = values.Encode()
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("creating http request: %w", err)
+	}
+	headers.SetUserAgent(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("doing http request: %w", err)
+	}
+	defer response.Body.Close()
+
+	b, err := io.ReadAll(response.Body)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("reading response body: %w", err)
+	}
+	s := string(b)
+
+	switch response.StatusCode {
+	case http.StatusOK:
+		switch {
+		case strings.Contains(s, "good"):
+			fmt.Println("GOOD!")
+			newIP, err = netip.ParseAddr(ip.String())
+			if err != nil {
+				return netip.Addr{}, fmt.Errorf("%w: %w", errors.ErrIPReceivedMalformed, err)
+			} else if !p.useProviderIP && ip.Compare(newIP) != 0 {
+				return netip.Addr{}, fmt.Errorf("%w: sent ip %s to update but received %s",
+					errors.ErrIPReceivedMismatch, ip, newIP)
+			}
+			return newIP, nil
+		case strings.Contains(s, "nochg"):
+			fmt.Println(s)
+			newIP, err = netip.ParseAddr(ip.String())
+			if err != nil {
+				return netip.Addr{}, fmt.Errorf("%w: in response %q", errors.ErrReceivedNoResult, s)
+			} else if !p.useProviderIP && ip.Compare(newIP) != 0 {
+				return netip.Addr{}, fmt.Errorf("%w: sent ip %s to update but received %s",
+					errors.ErrIPReceivedMismatch, ip, newIP)
+			}
+			return newIP, nil
+		default:
+			return netip.Addr{}, fmt.Errorf("%w: %s", errors.ErrUnknownResponse, s)
+		}
+	case http.StatusBadRequest:
+		switch s {
+		case constants.Nohost:
+			return netip.Addr{}, fmt.Errorf("%w", errors.ErrHostnameNotExists)
+		case constants.Badauth:
+			return netip.Addr{}, fmt.Errorf("%w", errors.ErrAuth)
+		default:
+			return netip.Addr{}, fmt.Errorf("%w: %d: %s", errors.ErrHTTPStatusNotValid, response.StatusCode, s)
```

Comment:
since we don't know how to handle the returned response, let's use the unknown response error instead

```suggestion
			return netip.Addr{}, fmt.Errorf("%w: %s", errors.ErrUnknownResponse, s)
```

Reply:
Corrected per your suggestion.